Makefile.kokkos

```diff
@@ -28,6 +28,8 @@ KOKKOS_OPTIONS ?= ""
 KOKKOS_CMAKE ?= "no"
 KOKKOS_TRIBITS ?= "no"
 KOKKOS_STANDALONE_CMAKE ?= "no"
+KOKKOS_ENABLE_CUDA_ATOMIC_INTRINSICS ?= 1
+KOKKOS_ENABLE_COLLAPSE_HIERARCHICAL_CONSTRUCTS ?= 1
```

Comment:
On OpenACC back end, we can use either CUDA intrinsic atomic operations or OpenACC annotations.
I think that right now, CUDA provides more compatibility with Kokkos atomics than OpenACC.
Maybe, it would be better to re-name these macros by specifing that those correspond to OpenACC back end only??
Maybe using KOKKOS_INTERNAL_USE_OPENACC_XXX, and move them to a "OpenACC" part??